pyvista/core/filters/image_data.py

```diff
@@ -1915,3 +2102,143 @@ def label_connectivity(
+        to automatically provide a result with the proper dimensions. Raise errors if the
+        desired output cannot be obtained.
+
+        .. versionadded:: 0.45.0
+
+        Parameters
+        ----------
+        operation_mask : bool, VectorLike[bool], Literal['0D', '1D', '2D', '3D', 'preserve']
+            The desired mask to control whether to resize dimensions.
+
+            - ``True``: all dimensions are modified.
+            - ``False``: no dimensions are modified.
+            - Can be specified as a sequence of 3 boolean allow modification on a per
+            dimension basis.
+            - ``'0D'``: convenience alias to output a 0D ImageData, if possible, i.e.,
+            dims will be similar to (1, 1, 1).
+            - ``'1D'``: convenience alias to output a 1D ImageData, if possible, i.e.,
+            dims will be similar to (>1, 1, 1).
+            - ``'2D'``: convenience alias to output a 2D ImageData, if possible, i.e.,
+            dims will be similar to (>1, >1, 1).
+            - ``'3D'``: convenience alias to output a 3D ImageData, if possible, i.e.,
+            dims will be similar to (>1, >1, >1).
+            - ``'preserve'``: convenience alias to not modify singleton dimensions.
+
+        operator: Callable
+            The operation that will be perform on the dimensions. Must be a :module:`~operator`.
+
+        operation_size : int, VectorLike[int]
+            The size of the operation, applied to all dimensions if specified as a ``int``
+            or applied on a per dimension basis.
+
+        Returns
+        -------
+        NDArray[bool]
+            A (3, ) shaped mask array that indicates which dimensions will be modified.
+
+        NDArray[int]
+            A (3, ) shaped array that with the new ImageData dimensions after applying
+            the operation.
+
+        Examples
+        --------
+        Get the dimensions on which to operate to obtain a 2D output while adding ``2``.
+
+        >>> import pyvista as pv
+        >>> import operator
+        >>> image = pv.ImageData(dimensions=(4, 1, 4))
+        >>> image._validate_dim_operation(
+        ...     operation_mask='2D',
+        ...     operator=operator.add,
+        ...     operation_size=2,
+        ... )
+        (array([ True, False,  True]), array([6, 1, 6]))
+
+        """
+        dims = np.asarray(self.dimensions)  # type: ignore[attr-defined]
+        # Build an array of the operation size
+        operation_size = _validation.validate_array3(operation_size, reshape=True, broadcast=True)
+
+        if not isinstance(operation_mask, str):
+            # Build a bool array of the mask, cast to int first to be a valid input for validate_array3
+            dims_mask = np.asarray(operation_mask).astype(int)
+            dims_mask = _validation.validate_array3(
+                dims_mask, reshape=True, broadcast=True, dtype_out=bool
```

Comment:
We should require the input dtype to be bool rather than casting the output since any object can be converted to a bool.
```suggestion
                dims_mask, reshape=True, broadcast=True, must_have_dtype=bool
```